### ISSUE TITLE
regression: Gamezhero preloader no longer works

### ISSUE THREAD

**Issue by n0samu** [ruffle-rs/ruffle#14206]
### Describe the bug

Since nightly 2023-11-13, AVM2 games that use the Gamezhero preloader show only a loading spinner, and loading never completes. The error is:
```
ERROR ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "complete", class: flash.events::Event, ptr: 0x21ec1b1a230 }) to handler FunctionObject(FunctionObject { ptr: 0x21ec21c3460 }) : ReferenceError: Error #1069: Property com.gameallianz.engine.as3.modules.skins::default not found on global and there is no default value.
        at com.gameallianz.engine.as3.modules.skins::SkinManager/loadSkinMainAction()
        at com.gameallianz.engine.as3.modules.skins::SkinManager/handlerSiteListOnData()
        at com.gameallianz.engine.as3.modules.skins::SkinManager/handlerXMLComplete()
```

### Expected behavior

The games should load.

### Content Location

Affected games include [Where is My Beard](https://www.kongregate.com/games/gamezhero/where-is-my-beard), [Back to Alien Party](https://www.gamezhero.com/games/backtoalienparty), and [Catchy Orbit](https://www.newgrounds.com/portal/view/630301/format/flash?emulate=flash).

SWF link for Where is My Beard: https://chat.kongregate.com/gamez/0022/8466/live/wimb_secure.swf

### Affected platform

Desktop app

### Operating system

Windows 10

### Additional information

I've bisected this to commit 1501a858f6b8dea6a3ecb858dc5582d7a76debb4 (#13966).

**Comment by Lord-McSweeney** [comment]
Also #14007 and #14101.

**Comment by n0samu** [comment]
This seems to happen while loading the following XML file: `http://assets.gameallianz.com/api/xml/skins.xml`
Blocking external connections (for example with `--proxy http://localhost` allows the games to load again.

**Comment by DenisSokoloff** [comment]
Hi Everyone. 
I removed the external API (engine.dat).  All games with Gamezhero API use the internal engine and can have an old Gamezhero logo, alert messages, or other bugs, but the preloader looks like it is working OK. I tested on our best games:

https://www.gamezhero.com/games/alienslikemilk
https://www.gamezhero.com/games/braveboy

Happy New Year!

**Comment by n0samu** [comment]
Thank you! I'll let you know if we track down the cause of this bug and fix it, in case you want to restore the data file later on.

**Comment by n0samu** [comment]
Sadly this is *not* fixed by #14627.

**Comment by n0samu** [comment]
Fixed by #14801. @DenisSokoloff With the latest release of Ruffle, it should be fine to restore the Gamezhero API files now!

**Comment by DenisSokoloff** [comment]
> Fixed by #14801. @DenisSokoloff With the latest release of Ruffle, it should be fine to restore the Gamezhero API files now!

Done! Check on https://www.gamezhero.com/games/alienslikemilk - Well done!
